src/xkernel.cpp

```diff
@@ -61,26 +64,52 @@ namespace xeus
                      interpreter_ptr interpreter,
                      history_manager_ptr history_manager,
                      server_builder builder)
-        : m_config(config)
-        , m_user_name(user_name)
-        , p_interpreter(std::move(interpreter))
-        , p_history_manager(std::move(history_manager))
-        , m_builder(builder)
+        : m_config(config),
+          m_user_name(user_name),
+          p_interpreter(std::move(interpreter)),
+          p_history_manager(std::move(history_manager)),
+          m_builder(builder)
+    {
+    }
+
+    xkernel::xkernel(const std::string& user_name,
+                     interpreter_ptr interpreter,
+                     history_manager_ptr history_manager,
+                     server_builder builder)
+        : m_user_name(user_name),
+          p_interpreter(std::move(interpreter)),
+          p_history_manager(std::move(history_manager)),
+          m_builder(builder)
     {
     }
 
     xkernel::~xkernel() {}
 
-    void xkernel::start()
+    void xkernel::init()
     {
         m_kernel_id = new_xguid();
         m_session_id = new_xguid();
 
+        if (m_config.m_key.size() == 0)
+        {
+            std::string ascii = "0123456789abcdefghijklmnopqrstuvwxyz";
+
+            std::random_device r;
+            std::default_random_engine generator(r());
+            std::uniform_int_distribution<std::size_t> distribution(0, 35);
+
+            m_config.m_key.resize(33);
+            for (auto& ch: m_config.m_key)
+            {
+                ch = ascii[distribution(generator)];
+            }
+            m_config.m_key[8] = '-';
+        }
```

Comment:
Now that the config parameter is a constant parameter of the server builder, a local flag should be set to know if the kernel need to get its configuration updated after the server is built and the socket are all bound.